packages/expo-updates/ios/EXUpdates/EXUpdatesFileDownloader.m

```diff
@@ -0,0 +1,116 @@
+//  Copyright © 2018 650 Industries. All rights reserved.
+
+#import <EXUpdates/EXUpdatesFileDownloader.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+NSString * const kEXUpdatesFileDownloaderErrorDomain = @"EXUpdatesFileDownloader";
+NSTimeInterval const kEXUpdatesDefaultTimeoutInterval = 60;
+
+@interface EXUpdatesFileDownloader () <NSURLSessionDataDelegate>
+@end
+
+@implementation EXUpdatesFileDownloader
+
+- (void)downloadFileFromURL:(NSURL *)url
+                     toPath:(NSString *)destinationPath
+               successBlock:(EXUpdatesFileDownloaderSuccessBlock)successBlock
+                 errorBlock:(EXUpdatesFileDownloaderErrorBlock)errorBlock
+{
+  [self downloadDataFromURL:url successBlock:^(NSData * _Nonnull data, NSURLResponse * _Nonnull response) {
+    if ([[NSFileManager defaultManager] createFileAtPath:destinationPath contents:data attributes:nil]) {
+      successBlock(data, response);
+    } else {
+      errorBlock([NSError errorWithDomain:kEXUpdatesFileDownloaderErrorDomain code:-1 userInfo:@{NSLocalizedDescriptionKey: [NSString stringWithFormat:@"Could not write to path: %@", destinationPath]}], response);
+    }
+  } errorBlock:errorBlock];
+}
+
+- (void)downloadDataFromURL:(NSURL *)url
+               successBlock:(EXUpdatesFileDownloaderSuccessBlock)successBlock
+                 errorBlock:(EXUpdatesFileDownloaderErrorBlock)errorBlock
+{
+  NSURLSessionConfiguration *configuration = _urlSessionConfiguration ?: [NSURLSessionConfiguration defaultSessionConfiguration];
```

Comment:
Since `defaultSessionConfiguration` is declared as a property rather than a method, `.` is semantically more accurate than message-passing syntax (though, both compile to the same code):
```suggestion
  NSURLSessionConfiguration *configuration = _urlSessionConfiguration ?: NSURLSessionConfiguration.defaultSessionConfiguration;
```

https://developer.apple.com/documentation/foundation/nsurlsessionconfiguration/1411560-defaultsessionconfiguration?language=objc